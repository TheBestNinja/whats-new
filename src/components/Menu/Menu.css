.menu {
  display: flex;
  justify-content: space-around;
  list-style-type: none;
  margin-bottom: 10px;
}

.topic {
  font-size: 20px;
  cursor: pointer;
  font-weight: 300;
}

.topic:hover {
  color: #00dae1;
}