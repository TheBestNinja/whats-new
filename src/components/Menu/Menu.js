import React from 'react';
import './Menu.css'

const Menu = ({ selectTopic }) => {
  const navOptions = ['local', 'entertainment', 'health', 'science', 'technology']
  return (
    <ul className='menu'>{navOptions.map((topic, i) => <li className='topic' key={i} id={topic} topic={topic} onClick={() => selectTopic(topic)}>{topic.toUpperCase()}</li>)} </ul> 
  )
  
}

export default Menu;