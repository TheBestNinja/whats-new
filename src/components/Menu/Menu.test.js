import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu'

describe('Menu', () => {
  it('should match the snapshot with all topics displayed', () =>{
    // setup
    const mockSelectTopic = jest.fn();

    // execution
    const wrapper = shallow(<Menu selectTopic={mockSelectTopic} />)

    // expectation
    expect(wrapper).toMatchSnapshot();
  })

  it('should be able to click a topic and have a function run', () => {
    // setup
    const mockSelectTopic = jest.fn();

    // execution
    const wrapper = shallow(<Menu selectTopic={mockSelectTopic} />)
    wrapper.find('li').at(0).simulate('click')
    
    // expectation 
    expect(mockSelectTopic).toHaveBeenCalled();

  })
})