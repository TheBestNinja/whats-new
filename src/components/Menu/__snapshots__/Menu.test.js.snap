// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all topics displayed 1`] = `
<ul
  className="menu"
>
  <li
    className="topic"
    id="local"
    key="0"
    onClick={[Function]}
    topic="local"
  >
    LOCAL
  </li>
  <li
    className="topic"
    id="entertainment"
    key="1"
    onClick={[Function]}
    topic="entertainment"
  >
    ENTERTAINMENT
  </li>
  <li
    className="topic"
    id="health"
    key="2"
    onClick={[Function]}
    topic="health"
  >
    HEALTH
  </li>
  <li
    className="topic"
    id="science"
    key="3"
    onClick={[Function]}
    topic="science"
  >
    SCIENCE
  </li>
  <li
    className="topic"
    id="technology"
    key="4"
    onClick={[Function]}
    topic="technology"
  >
    TECHNOLOGY
  </li>
   
</ul>
`;
